### ISSUE TITLE
Wi-Fi Available in Flight Mode

### ISSUE THREAD

**Issue by MarkMuth** [HelloVolla/ubuntu-touch-beta-tests#80]
Wi-Fi can be turned on and works just fine in Flight Mode. This is not quite consistent. See syslog and screenshot attached:

    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: handle_flight_mode_killswitch: killswitch[WLAN] state: UNBLOCKED block: 1
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: Setting WLAN devices to blocked
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: set_soft: Setting WLAN to blocked
    Dec  5 00:46:59 ubuntu-phablet wpa_supplicant[15541]: wlan0: CTRL-EVENT-DISCONNECTED bssid=c8:0e:14:2b:d3:59 reason=3 locally_generated=1
    Dec  5 00:46:59 ubuntu-phablet dhclient[26918]: receive_packet failed on wlan0: Network is down
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: device_changed_cb: phy0
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: urf_killswitch_soft_block_cb: Setting device /org/freedesktop/URfkill/devices/0 to blocked
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: set_soft: Setting WLAN to blocked
    Dec  5 00:46:59 ubuntu-phablet wpa_supplicant[15541]: rfkill: WLAN soft blocked
    Dec  5 00:46:59 ubuntu-phablet wpa_supplicant[15541]: rfkill: WLAN soft blocked
    Dec  5 00:46:59 ubuntu-phablet URfkill[832]: device_changed_cb: phy1
    Dec  5 00:46:59 ubuntu-phablet NetworkManager[2228]: <info>  [1607125619.5902] manager: WiFi now disabled by radio killswitch
    Dec  5 00:46:59 ubuntu-phablet NetworkManager[2228]: <info>  [1607125619.5906] device (wlan0): state change: activated -> unavailable (reason 'none') [100 20 0]
    Dec  5 00:46:59 ubuntu-phablet NetworkManager[2228]: <info>  [1607125619.5977] dhcp4 (wlan0): canceled DHCP transaction, DHCP client pid 26918
    Dec  5 00:46:59 ubuntu-phablet NetworkManager[2228]: <info>  [1607125619.5982] dhcp4 (wlan0): state changed bound -> done
    
    
    Dec  5 00:47:02 ubuntu-phablet kernel: [139459.048579] [MTK-WIFI] WIFI_open[I]: major 153 minor 0 (pid 4529)
    Dec  5 00:47:02 ubuntu-phablet kernel: [139459.050576] [MTK-WIFI] WIFI_write[I]: WIFI_write 1, length 1
    Dec  5 00:47:02 ubuntu-phablet kernel: [139459.050611] [MTK-WIFI] WIFI_write[I]: WIFI is already power on!
    Dec  5 00:47:02 ubuntu-phablet kernel: [139459.050989] [MTK-WIFI] WIFI_close[I]: major 153 minor 0 (pid 4529)
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: Setting WLAN devices to unblocked
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: set_soft: Setting WLAN to unblocked
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: device_changed_cb: phy0
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.6685] manager: WiFi now enabled by radio killswitch
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: urf_killswitch_soft_block_cb: Setting device /org/freedesktop/URfkill/devices/0 to unblocked
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: set_soft: Setting WLAN to unblocked
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: device_changed_cb: phy1
    Dec  5 00:47:02 ubuntu-phablet kernel: [139459.107435] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
    Dec  5 00:47:02 ubuntu-phablet URfkill[832]: urf_killswitch_soft_block_cb: all done
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.7326] audit: op="radio-control" arg="wireless-enabled:1" pid=4529 uid=32011 result="success"
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.7433] sup-iface[0xf5b6e20,wlan0]: supports 5 scan SSIDs
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.7501] device (wlan0): supplicant interface state: starting -> ready
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.7505] device (wlan0): state change: unavailable -> disconnected (reason 'supplicant-available') [20 30 42]
    Dec  5 00:47:02 ubuntu-phablet NetworkManager[2228]: <info>  [1607125622.7528] policy: Device 'wlan0' has no connection; scheduling activate_check in 0 seconds.
    

![WiFiInFlightMode](https://user-images.githubusercontent.com/4218151/101226563-25f87900-3695-11eb-80ec-42f410fde0ad.png)

**Comment by JamiKettunen** [comment]
I'm not sure what exactly is being reported here, but if it's that Wi-Fi can still be enabled while Flight Mode is enabled here's my take:

I'm pretty certain this is intended behavior; when you toggle Flight Mode on, all radio devices get disabled (if they weren't already) and return to their previous state when you toggle the mode off again. Wi-Fi and Bluetooth may still manually be toggled back on since *I think* many airlines these days also allow such features, it's just the cellular modem that can cause issues while aboard.

This exact same behavior can also be seen on any Android based OSes (and most likely iOS as well).

**Comment by JamiKettunen** [comment]
As I haven't heard anything back in 4 days I'll go ahead and close this issue. Please do comment here or make a new issue as needed if there's something on your mind :)
